packages/material-ui-based-components/src/components/Number.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useCallback } from 'react';
+import { withRuleEngine } from '../shared/withRuleEngine';
+import { PROPS } from '../utils/types';
+import { FormControl, FormHelperText, InputLabel, OutlinedInput } from '@mui/material';
+
+const NumberComponent = (props: PROPS) => {
+  const { isError, placeholder, name, required, label, description,
+    enabled, readOnly, maximum, minimum, errorMessage, id, value
+  } = props;
+
+  const changeHandler = useCallback((event: any) => {
+    props.dispatchChange(event.target.value);
+  }, [props.dispatchChange]);
+
+  const blurHandler = useCallback((event: any) => {
+    props.dispatchBlur(event.target.value);
+  }, [props.dispatchBlur]);
+
+  const focusHandler = useCallback((event: any) => {
+    props.dispatchFocus(event.target.value);
+  }, [props.dispatchFocus]);
+
+  return (
+    <FormControl
+      variant="outlined"
```

Comment:
variant should be dynamic. It will come from the props.properties['afs:layout'].variant and default value of this outlined

Reply:
Added variant